tests/unittests/Types_test.cpp

```diff
@@ -130,6 +138,16 @@ TEST(AVariant, ReturnsTrueWhenAskedIfItContainsTheTypeItReallyContains)
     ASSERT_TRUE(variant.containsValueOfType<ComplexType>());
 }
 
+TEST(AVariant, CanBeConvertedIntoAnStdVariant)
+{
+    using VariantType = std::variant<std::string, uint64_t, std::vector<sdbus::Struct<std::string, double>>>;
+    VariantType value{ std::vector<sdbus::Struct<std::string, double>>{{"hello"s, ANY_DOUBLE}, {"world"s, ANY_DOUBLE}} };
+
+    sdbus::Variant variant(value);
+
+    ASSERT_EQ(value, VariantType { variant });
+}
+
```

Comment:
Thinking more about it, I would suggest we split this test in two. One would test and demonstrate that we can do the conversion, the other would test and demonstrate that the two variant types are implicitly interchangeable (given my first comment above):

```suggestion
TEST(AVariant, CanBeConvertedIntoAnStdVariant)
{
    std::vector<sdbus::Struct<std::string, double>> value{{"hello"s, ANY_DOUBLE}, {"world"s, ANY_DOUBLE}};
    using StdVariantType = std::variant<std::string, uint64_t, std::vector<sdbus::Struct<std::string, double>>>;
    sdbus::Variant sdbusVariant(value);

    StdVariantType stdVariant{sdbusVariant};

    ASSERT_TRUE(std::holds_alternative<std::vector<sdbus::Struct<std::string, double>>>(stdVariant));
    ASSERT_THAT(std::get<std::vector<sdbus::Struct<std::string, double>>>(stdVariant), Eq(value));
}

TEST(AVariant, IsImplicitlyInterchangeableWithStdVariant)
{
    std::vector<sdbus::Struct<std::string, double>> value{{"hello"s, ANY_DOUBLE}, {"world"s, ANY_DOUBLE}};
    using StdVariantType = std::variant<std::string, uint64_t, std::vector<sdbus::Struct<std::string, double>>>;
    StdVariantType stdVariant{value};

    auto stdVariantCopy = [](const sdbus::Variant& v) -> StdVariantType{ return v; }(stdVariant);

    ASSERT_THAT(stdVariantCopy, Eq(stdVariant));
}

```